### PR TITLE
Add CFL subtract average test

### PR DESCRIPTION
**Pull request by edmond-zhu**
1. Add test for CFL subtract average functions

### CODE REVIEW

File: test/subtract_avg_cfl_test.cc [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+class CflSubAvgTest : public ::testing::TestWithParam<sub_avg_param> {
+  public:
+    CflSubAvgTest() : tx_size_(TEST_GET_PARAM(0)) {
+        width_ = tx_size_wide[tx_size_];
+        height_ = tx_size_high[tx_size_];
+        memset(data_, 0, sizeof(data_));
+        memset(data_ref_, 0, sizeof(data_ref_));
+
+        /** get test and reference function */
+        sub_avg = TEST_GET_PARAM(1)(tx_size_);
```
Comment (Cidana-Zhengwen) [inline test/subtract_avg_cfl_test.cc]: If get_subtract_average_fn_avx2() is the only test target, we can assign function ptr by get_subtract_average_fn_avx2 directly. And remove function out of MAKE_TEST_PARAM macro.

File: test/subtract_avg_cfl_test.cc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            for (int i = 0; i < width_; i++) {
+                data_[j * CFL_BUF_LINE + i] = data_ref_[j * CFL_BUF_LINE + i] =
+                    (int16_t)rnd.random();
+            }
+        }
+    }
+    void compare_data() {
+        for (int j = 0; j < height_; j++) {
+            for (int i = 0; i < width_; i++) {
+                ASSERT_EQ(data_[j * CFL_BUF_LINE + i],
```
Comment (Cidana-Zhengwen) [inline test/subtract_avg_cfl_test.cc]: If the test fail, could we dump the input data to reproduce this fail?

**Review by Cidana-Zhengwen — COMMENTED** [review]
Some suggestions, maybe helpful.

File: test/subtract_avg_cfl_test.cc (hunk 1 above)
Comment (edmond-zhu) [reply test/subtract_avg_cfl_test.cc]: it's done

File: test/subtract_avg_cfl_test.cc (hunk 2 above)
Comment (edmond-zhu) [reply test/subtract_avg_cfl_test.cc]: it's done.

**Review by Cidana-Zhengwen — APPROVED** [review]
Looks good.

**Comment by Cidana-Developers** [comment]
merged，close it.

**Review by Cidana-Developers — APPROVED** [review]

**Pull request closed without merging**